src/xcode/ENA/ENA/Source/Services/Exposure Transaction/__tests__/ExposureDetectionExecutorTests.swift

```diff
@@ -0,0 +1,393 @@
+				XCTAssertEqual(summary.matchedKeyCount, mockSummary.matchedKeyCount)
+				XCTAssertEqual(summary.maximumRiskScore, mockSummary.maximumRiskScore)
+			}
+		)
+		waitForExpectations(timeout: 2.0)
+	}
+
+	func testDetectSummaryWithConfiguration_Error() throws {
+		// Test the case where the exector is asked to run an exposure detection
+		// We provide an `MockExposureDetector` with an error, and expect this to be returned
+		let completionExpectation = expectation(description: "Expect that the completion handler is called.")
+		let expectedError = ENError(.notAuthorized)
+		let sut = ExposureDetectionExecutor.makeWith(exposureDetector: MockExposureDetector((nil, expectedError)))
+
+		sut.exposureDetection(
+			ExposureDetection(delegate: sut),
+			detectSummaryWithConfiguration: ENExposureConfiguration(),
+			writtenPackages: WrittenPackages(urls: []),
+			completion: { result in
+				defer { completionExpectation.fulfill() }
+
+				guard case .failure(let error) = result else {
+					XCTFail("Completion handler indicated succeess though it should have failed!")
+					return
+				}
+
+				guard let enError = error as? ENError else {
+					XCTFail("Completion handler returned incorrect error type. Expected ENErrror, got \(type(of: error))!")
+					return
+				}
+
+				XCTAssertEqual(enError.code, expectedError.code)
+			}
+		)
+		waitForExpectations(timeout: 2.0)
+	}
+}
+
+// MARK: - Private Helper Extensions
+
+private extension ExposureDetectionExecutor {
+	/// Create a `ExposureDetectionExecutor` with the specified parameters. Mock defaults are applied unless specified.
+	static func makeWith(
+		client: Client = ClientMock(),
+		packageStore: DownloadedPackagesStore = DownloadedPackagesSQLLiteStore.inMemory(),
+		store: Store = MockTestStore(),
+		exposureDetector: MockExposureDetector = MockExposureDetector()
+	) -> ExposureDetectionExecutor {
+		ExposureDetectionExecutor(
+			client: client,
+			downloadedPackagesStore: packageStore,
+			store: store,
+			exposureDetector: exposureDetector
+		)
+	}
+}
+
+private extension DownloadedPackagesSQLLiteStore {
+	static var openInMemory: DownloadedPackagesSQLLiteStore {
+		let store = DownloadedPackagesSQLLiteStore.inMemory()
+		store.open()
+		return store
+	}
+}
+
+private extension Date {
+	var formatted: String {
+		DateFormatter.packagesDateFormatter.string(from: self)
+	}
+}
+
+private extension Calendar {
+	static var gregorianUTC: Calendar {
+		var calendar = Calendar(identifier: .gregorian)
+		// swiftlint:disable:next force_unwrapping
+		calendar.timeZone = TimeZone(abbreviation: "UTC")!
+		return calendar
+	}
+}
+
+private extension DateFormatter {
+	static var packagesDateFormatter: DateFormatter = {
+		let formatter = DateFormatter()
+		formatter.dateFormat = "yyyy-MM-dd"
+		formatter.timeZone = TimeZone(abbreviation: "UTC")
+		formatter.locale = Locale(identifier: "en_US_POSIX")
+		formatter.calendar = Calendar(identifier: .gregorian)
+		return formatter
+	}()
+}
```

Comment:
This is a straight copy and paste from `String+Today.swift`
The alternative would be to add this file to the unit test target as well.